src/pathwayvis/components/map/map.component.ts

```diff
@@ -330,16 +327,35 @@ class MapComponentCtrl {
   /**
    * Callback function for clicked action button in context menu
    */
-  public processActionClick(action, data) {
-
+  public processActionClick(action, data, element) {
     if (action.type === 'reaction:link') {
       this.$window.open(`http://bigg.ucsd.edu/universal/reactions/${data.bigg_id}`);
+    } else if (action.type.startsWith('reaction:objective')) {
+      this.shared.async(this._mapOptions.actionHandler(action, { id: data.bigg_id }).then((response) => {
+        if (action.type.endsWith('do')) {
+            this._mapOptions.setCurrentGrowthRate(parseFloat(response['growth-rate']));
+            this._mapOptions.setReactionData(response.fluxes);
+            this._mapOptions.setRemovedReactions(response['removed-reactions']);
+            this._mapOptions.setObjectiveReaction(data.bigg_id);
+            this._getContext();
+        } else {
+            this._mapOptions.setObjectiveReaction("");
+            this._loadMap(this._mapOptions.getType(), this._mapOptions.getDataObject().selected, this._mapOptions.getSelectedId());
+        }
+      }, (error) => {
+        this.toastService.showErrorToast('Oops! Sorry, there was a problem.');
+      }), 'Objective');
     } else {
-      this._mapOptions.actionHandler(action, { id: data.bigg_id }).then((response) => {
+      this.shared.async(this._mapOptions.actionHandler(action, { id: data.bigg_id }).then((response) => {
```

Comment:
Could you also merge this one to here? https://github.com/DD-DeCaF/metabolica-ui-map/pull/141/files#diff-62c06d0a48047fe88322f20be6e856e2R334